user-data-cleanup/functions/access.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const exp = require('./expression');
+
+/**
+ * Access Class, used to represent the access status of a write rule or node
+ * @param status access status, could be NO_ACCESS/SINGLE_ACCESS/MULT_ACCESS
+ * @param list variable list, should be empty list if status is NO/SINGLE,
+ * else should be the list of literal in the conjunction of corresponding exp
+ */
+function Access(status, list) {
```

Comment:
Just FYI, since this runs on Functions and Functions run on Node 6 we could have used the ES2015 `class` notation here. I'd say this is fine though but if you want/have time to re-write in class notation it would make it a bit easier to read :) (I know there isn't that much time left so it's fine to leave as-is also :))